inst/include/child_model_simulation.hpp

```diff
@@ -13,54 +13,70 @@ void run_child_ageing(int time_step,
                       const State<ModelVariant, real_type> &state_curr,
                       State<ModelVariant, real_type> &state_next,
                       IntermediateData<ModelVariant, real_type> &intermediate) {
-  const auto demog = pars.base.demography;
-  const auto cpars = pars.children.children;
+  const auto survival_probability = pars.base.demography.survival_probability;
+  const auto hc_cd4_transition = pars.children.children.hc_cd4_transition;
+  const auto p_idx_fertility_first = pars.base.options.p_idx_fertility_first;
+
+  const auto hc1_hiv_pop_curr = state_curr.children.hc1_hiv_pop;
+  const auto hc1_art_pop_curr = state_curr.children.hc1_art_pop;
+  const auto hc2_hiv_pop_curr = state_curr.children.hc2_hiv_pop;
+  const auto hc2_art_pop_curr = state_curr.children.hc2_art_pop;
+
+  auto hc1_hiv_pop_next = state_next.children.hc1_hiv_pop;
+  auto hc1_art_pop_next = state_next.children.hc1_art_pop;
+  auto hc2_hiv_pop_next = state_next.children.hc2_hiv_pop;
+  auto hc2_art_pop_next = state_next.children.hc2_art_pop;
+
   static_assert(ModelVariant::run_child_model,
                 "run_hiv_child_infections can only be called for model variants where run_child_model is true");
-  constexpr auto ss = StateSpace<ModelVariant>().base;
-  constexpr auto hc_ss = StateSpace<ModelVariant>().children;
-
-  for (int s = 0; s < ss.NS; ++s) {
+  constexpr StateSpace<ModelVariant> ss = StateSpace<ModelVariant>();
+  constexpr auto NS = ss.base.NS;
+  constexpr auto hTS = ss.base.hTS;
+  constexpr auto hc2_agestart = ss.children.hc2_agestart;
+  constexpr auto hc1DS = ss.children.hc1DS;
+  constexpr auto hc2DS = ss.children.hc2DS;
+  constexpr auto hcTT = ss.children.hcTT;
+  constexpr auto hc1_ageend = ss.children.hc1_ageend;
+
+  for (int s = 0; s < NS; ++s) {
     //less than 5 because there is a cd4 transition between ages 4 and 5
-    for (int a = 1; a < hc_ss.hc2_agestart; ++a) {
-      for (int hd = 0; hd < hc_ss.hc1DS; ++hd) {
-        for (int cat = 0; cat < hc_ss.hcTT; ++cat) {
-          state_next.children.hc1_hiv_pop(hd, cat, a, s) +=
-            state_curr.children.hc1_hiv_pop(hd, cat, a - 1, s) * demog.survival_probability(a, s, time_step);
+    for (int a = 1; a < hc2_agestart; ++a) {
+      const auto survival_prob_per_group = survival_probability(a, s, time_step);
```

Comment:
I don't like this change; prefer keeping the indexes referenced in the line below to easily parse what the calculation is.

i.e., I prefer: 
```
hc1_hiv_pop_next(hd, cat, a, s) += hc1_hiv_pop_curr(hd, cat, a - 1, s) * survival_probability(a, s, time_step)
```
instead of
```
hc1_hiv_pop_next(hd, cat, a, s) += hc1_hiv_pop_curr(hd, cat, a - 1, s) * survival_prob_per_group
```

If we want to make these more concise for readability, I would be open to abbreviating `time_step` to something shorter.

Reply:
I agree with this. Switching `time_step` to `ts` or similar in the function signature sounds sensible.

Reply:
sounds good, then for now i wont be introducing any proxy variables (would you like proxy variables for multiple tensor expression, for example [this](https://github.com/mrc-ide/frogger/pull/107/files#diff-563821e9ce79549a4ac984cd4b62632d6659a4dc04fbc27aebd9c52b414477d1R59), can talk about it in the meeting), also will be abbreviating `time_step` to `ts`